src/createRollup.integration.test.ts

```diff
@@ -1,56 +1,121 @@
 import { describe, it, expect } from 'vitest';
-import { createPublicClient, http, parseGwei, zeroAddress } from 'viem';
+import { Address, createPublicClient, http, parseGwei, zeroAddress } from 'viem';
 
 import { nitroTestnodeL2 } from './chains';
 import { getNitroTestnodePrivateKeyAccounts } from './testHelpers';
 import { createRollupFetchTransactionHash } from './createRollupFetchTransactionHash';
-import { createTestRollup } from './utils/testHelpers';
+import { generateChainId } from './utils';
+import { createRollup } from './createRollup';
+import { createRollupPrepareConfig } from './createRollupPrepareConfig';
+import { prepareChainConfig } from './prepareChainConfig';
 
-// public client
-const publicClient = createPublicClient({
+const parentChainPublicClient = createPublicClient({
   chain: nitroTestnodeL2,
   transport: http(),
 });
 
 // test inputs
 const testnodeAccounts = getNitroTestnodePrivateKeyAccounts();
-const deployer = testnodeAccounts.deployer;
+const userTokenBridgeDeployer = testnodeAccounts.userTokenBridgeDeployer;
 const batchPoster = testnodeAccounts.deployer.address;
 const validators = [testnodeAccounts.deployer.address];
 
-describe(`createRollup`, async () => {
-  // create test rollup
-  const createRollupInformation = await createTestRollup({
-    deployer,
+async function createRollupHelper(nativeToken: Address = zeroAddress) {
+  const chainId = generateChainId();
+
+  const createRollupConfig = createRollupPrepareConfig({
+    chainId: BigInt(chainId),
+    owner: userTokenBridgeDeployer.address,
```

Comment:
we need to add it because it was the only account that received the custom token deployed by the testnode script from the `--l3-fee-token` flag

it's named `userTokenBridgeDeployer` because it's the `user_token_bridge_deployer` from nitro testnode